src/js/api.ts

```diff
@@ -378,6 +378,15 @@ export function unpackArchive(
     }
     options = { extractDir: options };
   }
+  if (ArrayBuffer.isView(buffer)) {
+    buffer = buffer.buffer;
+  }
```

Comment:
If I understand correctly if it's a TypedArray it's converted to ArrayBuffer here , and then on the following line back to TypedArray.

So that a roundtrip for TypedArray input. Can't we only convert ArrayBuffer to  TypedArray and keep the TypedArray unchanged?

Reply:
Ideally we always want to pass `Uint8Array`. I'm concerned that the Emscripten FS APIs misbehave otherwise.